### PR TITLE
Added an autochip flag to the MovePrimitive

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

This PR adds an autochip flag to the MovePrimitive, similar to the existing autokick. This allows the robot to chip when the breakbeam is fired. The autokick / autochip now uses enum flags to indicate which to do, since both can't be done at the same time.

_**This will require flashing firmware in order to test**_

### Testing Done

Unit tests all pass. Needs field testing.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/firmware/test/test.c [hunk 1]
```diff
@@ -7,60 +7,28 @@
...
-    run_physics_test();
-    run_quadratic_test();
-    run_shoot_test();
-    run_util_test();
  * when the main function is called each test is run.
-    (number_failed == 0) ? printf("All tests passed.\n") : printf("%d Tests failed.\n\n", number_failed);
  */
-    return 0;
 int main(void)
-}
```
Comment (garethellis0) [inline src/thunderbots/firmware/test/test.c:63]: What's up with the changes to this file?

File: src/thunderbots/software/ai/hl/stp/action/move_action.h [hunk 2]
```diff
@@ -33,16 +34,16 @@ class MoveAction : public Action
      * the destination
      * @param final_speed The final speed the robot should have at the destination
      * @param enable_dribbler Whether or not to enable the dribbler
-     * @param enable_autokick This will enable the "break-beam" on the robot, that will
-     *                        trigger the kicker to fire as soon as the ball is in front
-     *                        of it
+     * @param autokick This will enable the "break-beam" on the robot, that will
+     *    *                 trigger the kicker to fire as soon as the ball is in front of
+     * it
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/action/move_action.h]: formatting, also we should probably say "trigger the chipper or kicker"

File: src/thunderbots/software/ai/primitive/move_primitive.h [hunk 3]
```diff
@@ -9,6 +9,13 @@
 #include "geom/angle.h"
 #include "geom/point.h"
 
+enum AutokickFlag
```
Comment (garethellis0) [inline src/thunderbots/software/ai/primitive/move_primitive.h]: Can we rename to `AutokickType` since flags are usually binary?

Comment (garethellis0) [reply src/thunderbots/software/ai/primitive/move_primitive.h]: Same goes for functions like `getAutokickFlag`

**Review by garethellis0 — APPROVED** [review]
No logic problems, just naming/formatting

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Whoops did not mean to approve

File: src/thunderbots/firmware/test/test.c (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/firmware/test/test.c:63]: Not totally sure, but the changes constantly appear during formatting and other operations. The result seems fine, and it won't affect any behavior so I figured I'd just put it through so it stops complaining.

File: src/thunderbots/software/ai/hl/stp/action/move_action.h (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/action/move_action.h]: fixed

File: src/thunderbots/software/ai/primitive/move_primitive.h (hunk 3 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/primitive/move_primitive.h]: good idea. fixed

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Review by garethellis0 — APPROVED** [review]

**Pull request merged**